changefeedtrigger/__init__.py

```diff
@@ -0,0 +1,94 @@
+import time
+import json
+from subprocess import Popen, PIPE
+from azure.storage.blob.changefeed import ChangeFeedClient
+from azure.storage.blob import BlobServiceClient
+import azure.functions as func
+from datetime import timedelta, datetime
+import logging
+from predicate import where
+
+
+def multidimensional_filtering(cf_client, b_filter, blob_client, cont_name, c_filter, e_filter):
+    filter_event_service = blob_client.get_container_client(container=cont_name)
+    blob_name = "continuation-token"
+
+    # access blobs in container
+    blob_list = filter_event_service.list_blobs()
+    tokens = []
+    for blob in blob_list:
+        tokens.append(blob.name)
+    
+    # check if continuation token already exists
+    if blob_name in tokens:
+        cursor = filter_event_service.download_blob(blob_name).readall()
+        # if value of token is None return None
+        if cursor == b"None":
+            return None
+        # if value of token is a cursor initiate the changefeed with cursor
+        elif cursor != b"" and cursor != b"None":
+            change_feed = cf_client.list_changes(results_per_page=500).by_page(continuation_token=eval(cursor))
+        # if token is empty initiate changefeed without cursor
+        else:
+            change_feed = cf_client.list_changes(results_per_page=500).by_page()
+    # if token does not exist upload token as empty string placeholder as blob and initiate changefeed without cursor
+    else:
+        filter_event_service.upload_blob(blob_name, "") 
+        change_feed = cf_client.list_changes(results_per_page=500).by_page()
+
+    # iterate through change feed
+    change_feed_page = next(change_feed)
+
+    # get multidimensionally filtered events
+    #filtered_events = list(filter(where(data__eventType__iexact=e_filter, data__subject__icontains=("/" + c_filter + "/blobs/" + b_filter)), events))
+    filtered_events = []
+    for event in change_feed_page:
+        if (event["eventType"] == e_filter) and (("/" + c_filter + "/") in event["subject"]) and (("/" + b_filter) in event["subject"]):
+            filtered_events.append(event)
+        # set continuation token as last event in the filtered event list
+        cursor = change_feed.continuation_token
+    
```

Comment:
i separated it so that the user can comment out the method that filters it, so that they can just print out the entire changefeed log without filtering if they want to!